pandas/tests/io/parser/usecols.py

```diff
@@ -481,7 +481,7 @@ def test_raise_on_usecols_names_mismatch(self):
         data = 'a,b,c,d\n1,2,3,4\n5,6,7,8'
 
         if self.engine == 'c':
-            msg = 'Usecols do not match names'
+            msg = "do not match columns, columns expected but not found"
```

Comment:
Also, is this non-helpful error an issue when you pass in `engine='python'` ?  We would like to make sure that both engines are equally expressive about these types of errors.